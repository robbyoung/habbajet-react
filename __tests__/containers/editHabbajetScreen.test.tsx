import React from 'react';
import renderer from 'react-test-renderer';
import store from '../../app/store';
import {Provider} from 'react-redux';
import EditHabbajetScreen from '../../app/containers/editHabbajetScreen';

jest.mock('@fortawesome/react-native-fontawesome', () => ({
    FontAwesomeIcon: '',
}));

jest.mock('../../app/storage', () => ({
    saveState: () => undefined,
}));

describe('Edit Habbajet Screen Component', () => {
    it('will map state to a habbajet edit form', () => {
        const component = renderer.create(
            <Provider store={store}>
                <EditHabbajetScreen />
            </Provider>,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});
