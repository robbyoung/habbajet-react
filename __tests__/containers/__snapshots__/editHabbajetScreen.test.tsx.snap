// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit Habbajet Screen Component will map state to a habbajet edit form 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "height": "100%",
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        }
      }
    >
      <View
        style={
          Object {
            "marginBottom": 15,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#959595",
                "fontFamily": "Abel",
                "fontSize": 20,
                "margin": 0,
                "padding": 0,
              },
            ]
          }
        >
          Name
        </Text>
        <TextInput
          allowFontScaling={true}
          keyboardType="default"
          onChangeText={[Function]}
          rejectResponderTermination={true}
          style={
            Object {
              "borderColor": "#959595",
              "borderWidth": 1,
              "fontFamily": "Abel",
              "fontSize": 30,
              "paddingLeft": 10,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View />
      </View>
      <View
        style={
          Object {
            "marginBottom": 15,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#959595",
                "fontFamily": "Abel",
                "fontSize": 20,
                "margin": 0,
                "padding": 0,
              },
            ]
          }
        >
          Value
        </Text>
        <TextInput
          allowFontScaling={true}
          keyboardType="numeric"
          onChangeText={[Function]}
          placeholder="50"
          rejectResponderTermination={true}
          style={
            Object {
              "borderColor": "#959595",
              "borderWidth": 1,
              "fontFamily": "Abel",
              "fontSize": 30,
              "paddingLeft": 10,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View />
      </View>
      <View
        style={
          Object {
            "marginBottom": 15,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#959595",
                "fontFamily": "Abel",
                "fontSize": 20,
                "margin": 0,
                "padding": 0,
              },
            ]
          }
        >
          Modifier
        </Text>
        <TextInput
          allowFontScaling={true}
          keyboardType="numeric"
          onChangeText={[Function]}
          placeholder="2"
          rejectResponderTermination={true}
          style={
            Object {
              "borderColor": "#959595",
              "borderWidth": 1,
              "fontFamily": "Abel",
              "fontSize": 30,
              "paddingLeft": 10,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View />
      </View>
      <View
        style={
          Object {
            "marginBottom": 15,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#959595",
                "fontFamily": "Abel",
                "fontSize": 20,
                "margin": 0,
                "padding": 0,
              },
            ]
          }
        >
          Slack Days
        </Text>
        <TextInput
          allowFontScaling={true}
          keyboardType="numeric"
          onChangeText={[Function]}
          placeholder="0"
          rejectResponderTermination={true}
          style={
            Object {
              "borderColor": "#959595",
              "borderWidth": 1,
              "fontFamily": "Abel",
              "fontSize": 30,
              "paddingLeft": 10,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View />
      </View>
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#959595",
                "fontFamily": "Abel",
                "fontSize": 20,
                "margin": 0,
                "padding": 0,
              },
            ]
          }
        >
          Color
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "space-between",
              "marginBottom": 10,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#8066C2",
                "borderColor": "#959595",
                "borderRadius": 4,
                "borderStyle": "dotted",
                "borderWidth": 4,
                "height": 60,
                "marginBottom": 10,
                "marginRight": 10,
                "opacity": 1,
                "width": 60,
              }
            }
          />
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#E04747",
                "borderRadius": 4,
                "height": 60,
                "marginBottom": 10,
                "marginRight": 10,
                "opacity": 1,
                "width": 60,
              }
            }
          />
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#2AAD5E",
                "borderRadius": 4,
                "height": 60,
                "marginBottom": 10,
                "marginRight": 10,
                "opacity": 1,
                "width": 60,
              }
            }
          />
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#B1B952",
                "borderRadius": 4,
                "height": 60,
                "marginBottom": 10,
                "marginRight": 10,
                "opacity": 1,
                "width": 60,
              }
            }
          />
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#7F90EC",
                "borderRadius": 4,
                "height": 60,
                "marginBottom": 10,
                "marginRight": 10,
                "opacity": 1,
                "width": 60,
              }
            }
          />
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#37353C",
                "borderRadius": 4,
                "height": 60,
                "marginBottom": 10,
                "marginRight": 10,
                "opacity": 1,
                "width": 60,
              }
            }
          />
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#E64CE9",
                "borderRadius": 4,
                "height": 60,
                "marginBottom": 10,
                "marginRight": 10,
                "opacity": 1,
                "width": 60,
              }
            }
          />
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#EB712D",
                "borderRadius": 4,
                "height": 60,
                "marginBottom": 10,
                "marginRight": 10,
                "opacity": 1,
                "width": 60,
              }
            }
          />
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#959595",
            "borderRadius": 10,
            "marginBottom": 10,
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#ffffff",
                "fontFamily": "Abel",
                "fontSize": 30,
                "padding": 10,
                "textAlign": "center",
                "width": "100%",
              },
            ]
          }
        >
          Done
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
